### DISCUSSION TITLE
Очень много лишнего css в cdek-widget.umd.js v3.8.2

### DISCUSSION THREAD

**Discussion by nexoos-team** [cdek-it/widget#32] [Ideas]
Для каждого элемента генерируются однотипные стили, что сильно увеличивает размер кода и время его выполнения.
Еще я бы предложил сделать подключение css отдельным файлом.

**Comment by vermorag** [comment]
Добрый день. Есть момент с тем, что отдельный css может подгружаться долго/не подгрузиться. И как быть с этой ситуацией? Учитывая, что это внешний сайт - крайне сложно контролировать на нем подгрузку больше одного файла.
В целом, идея интересная, но как ее разворачивать в слабо контролируемом окружении не совсем понятно.

**Comment by piton-zel** [comment]
Странный аргумент что отдельный css может не подгрузиться, с чего вдруг? Интернет пропал? Сервер упал? 
По сути да, такое может произойти, а еще если следовать этой логике, то и js может не подгрузиться, здесь тогда какое будет решение? Как по мне, стили должны жить со стилями, тем более сейчас каждая вторая cms все стили компилирует в один файл (хоть и бесполезное действо для h2).
Проблема в другом, что при таком действии, это когда стили генерит js, автоматом сейчас ловятся следующие проблемы
1. При многоразовом подключении (точнее подгрузки js, а такое частенько вижу при динамических подгрузках) в html-е генерируются совершенно одинаковые блоки style, как пример, человек 2-3 нажал кнопку открыть карту и выбрать пункт выдачи, автоматом получаем 2-3 совершенно одинаковых блока со стилями.
Надо или проверку на существование данного блока добавить (чтобы их не дублировать), или просто вынести в нормальный css файл. Если очень хочется запихнуть все и вся в js файл, то тогда по destroy удаляйте и style, а если уходить в высший пилотаж, то тогда эти стили необходимо разместить в том же div в котором и находится сам виджет, я имею ввиду <style scoped>, чтобы при destroy подчищались и они тоже.
2. Почему вы так жестко ограничиваете разработчиков на изменение, оформления вашего виджета? Если бы стили жили отдельно, с нормальными названиями стилей, то это было бы достаточно просто вносить корректировки для подгонки виджета под разрабатываемый проект, сейчас надо дополнительно еще догружать поверх уже свой css, в котором и живут эти корректировки с кривыми именами. 
Банальное и самое распространенное, изменить шрифт виджета на шрифт клиента, по факту простым способом никак, т.к. в var это у вас не вынесено, как много чего и другого, редактировать это в js можно, но как-то не по фен-шую, а название классов при очередной вашей сборке скорее всего у вас изменятся.
Кстати, стили можно догрузить и с помощью js, причем по факту повторная загрузка уже будет происходить из кеша браузера.
